src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -219,14 +224,14 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                     // Only add a comma after the first param, if any
                     builder.Append(',');
                 }
-                builder.Append(" error=\"invalid_token\", error_description=\"");
-                builder.Append(errorDescription);
+                builder.Append($" error=\"{eventContext.Error ?? "invalid_token"}\", error_description=\"");
+                builder.Append(eventContext.ErrorDescription ?? "An error occurred when validating the access token.");
                 builder.Append('\"');
 
                 Response.Headers.Append(HeaderNames.WWWAuthenticate, builder.ToString());
             }
 
-            return false;
+            return true;
```

Comment:
Pretty sure having multiple values in the same `WWW-Authenticate` header is valid, but in practice, I'm not sure many clients will correctly parse it when trying to extract the `error`/`error_description`.